lib/cocoapods/installer.rb

```diff
@@ -196,23 +196,41 @@ def download_dependencies
 
     private
 
-    def create_generator
-      Xcode::PodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+    def create_generator(generate_multiple_pod_projects = false)
+      if generate_multiple_pod_projects
+        Xcode::MultiPodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+      else
+        Xcode::SinglePodsProjectGenerator.new(sandbox, aggregate_targets, pod_targets, analysis_result, installation_options, config)
+      end
     end
 
-    # Generate the 'Pods/Pods.xcodeproj' project.
+    # Generates the Xcode project(s) that go inside the `Pods/` directory.
     #
-    def generate_pods_project(generator = create_generator)
+    def generate_pods_project(generator = create_generator(installation_options.generate_multiple_pod_projects))
       UI.section 'Generating Pods project' do
         pod_project_generation_result = generator.generate!
         @target_installation_results = pod_project_generation_result.target_installation_results
         @pods_project = pod_project_generation_result.project
+        @pod_target_subprojects = pod_project_generation_result.projects_by_pod_targets.keys
```

Comment:
please add a comment here explaining why we need the `ivar` for backwards compatibility.